R/IDateTime.R

```diff
@@ -5,9 +5,14 @@
 
 as.IDate <- function(x, ...) UseMethod("as.IDate")
 
-as.IDate.default <- function(x, ..., tz = attr(x, "tzone")) {
+as.IDate.default <- function(x, ..., tz = attr(x, "tzone"), use_merge = 'auto') {
   if (is.null(tz)) tz = "UTC"
-  as.IDate(as.Date(x, tz = tz, ...))
+  if (isTRUE(use_merge) || (use_merge == 'auto' && length(x) >= 1000)) {
+    DT = data.table(x)
+    DT[unique(DT)[ , IDate := as.IDate(x, tz = tz, ...)], on = 'x', i.IDate]
+  } else {
+    as.IDate(as.Date(x, tz = tz, ...))
```

Comment:
Why the use of `as.IDate(x, tz = tz, ...)` when `use_merge` is TRUE as opposed to `as.IDate(as.Date(x, tz = tz, ...))` when `use_merge` is FALSE?

Reply:
because I wrote the code in different commits 😝 will fix...